programs/nft_api/src/nft_state.rs

```diff
@@ -0,0 +1,17 @@
+use bincode::serialize_into;
+use serde_derive::{Deserialize, Serialize};
+use solana_sdk::{instruction::InstructionError, pubkey::Pubkey, short_vec};
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default, PartialEq)]
+pub struct NftState {
+    pub issuer_pubkey: Pubkey,
+    pub owner_pubkey: Pubkey,
+    #[serde(with = "short_vec")]
+    pub id: Vec<u8>,
```

Comment:
Could this be `data`?  We already get uniqueness for free from the NFT being an account

Reply:
That's a good point. I'll play with abandoning both `issuer_pubkey` and `id`.  I was thinking of that field as either holding a device's ESN or a Cryptokitty's description, but I'd rather have the most minimal NFT before adding features.  Easy to imagine a separate account mapping pubkeys to different identifier formats. And if that was too slow, creating a new program to add that feature, accepting that it adds 33 bytes of overhead per account.